### ISSUE TITLE
Assorted issues with ComputerCraft support

### ISSUE THREAD

**Issue by LemADEC** [TeamPneumatic/pnc-repressurized#300]
### Minecraft Version
1.12.2

### Forge Version
n/a

### Mod Version
1.12.2-0.9.1-317

### Describe your problem, including steps to reproduce it
Issue A: Pneumaticraft registers a provider for (nearly) every blocks in the mod. In practice, you only need one. Current approach is adding unnecessary CPU load.

Issue B: As per CC API documentation for IPeripheralProvider.getPeripheral(), the latest should return null when it's not a supported block. As it's currently implemented, all blocks implementing IPeripheral (from CC API), including those from other mods, will be rerouted by PneumaticCraft IPeripheralProvider.

Nota: the combination of both issues causes all blocks implementing IPeripheral to be called 45 extra times for each attach/detach events from ComputerCraft API while Computronics is installed.

Issue C: most IPeripheral methods, including attach() & detach() are called from the related computer thread.
Since multiple computers can attach to the same block, the related data structures should be thread safe. 
Issue C': The `notifyComputers()` method is called from `update()` which is in the client/server thread. As such, it would also require thread protection when accessing the `attachedComputers` collections.

The following collection in PneumaticCraft isn't thread safe:
https://github.com/TeamPneumatic/pnc-repressurized/blob/fc83f32bd060798c42ea34e53f7257b8ff3df055/src/main/java/me/desht/pneumaticcraft/common/thirdparty/computercraft/TileEntityDroneInterface.java#L46
https://github.com/TeamPneumatic/pnc-repressurized/blob/3cfcb91e2e51953d03237e0a8d8c46bd85a44e70/src/main/java/me/desht/pneumaticcraft/common/tileentity/TileEntityUniversalSensor.java#L79

```java
private final List<IComputerAccess> attachedComputers = new ArrayList<>();
```
Consider using a CopyOnWriteArray collection instead, it's naturally thread safe and well suited for this kind of usage (many read, few write, small size).

### Any other comments?
There's probably a lag impact during chunk loading and unloading from those issues, I didn't profile to verify it's significant. For sure, fixing the above would prevent 'random' ConcurrentModificationExceptions.

**Comment by desht** [comment]
Thanks for the bug reports, there does appear to be some work to do here.  I didn't actually write the CC integration code, and I'm not overly familiar with the CC API.  Do you have a link to the docs you mentioned?

(Since you seem pretty familiar with how the code should work, I may have further questions for you.  Also, pull requests are always welcome :)

**Comment by desht** [comment]
Update: after a little code perusal, B & C look pretty simple to fix.

However, for A, what do you mean by "only need one"? As far as I can see, ``registerPeripheralProvider()`` needs to be called for each block that can be a CC peripheral, which is indeed most PneumaticCraft blocks (providing info about pressure and/or temperature are methods common to nearly all machines in the mod).

**Comment by desht** [comment]
Fixed in 0.9.2 release
